### PR TITLE
Add action/:id/execute endpoint

### PR DESCRIPTION
**Pull request by tsmacdonald**
BE for #28305

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28335)

### CODE REVIEW

File: src/metabase/actions/execution.clj [hunk 1]
```diff
@@ -162,12 +162,13 @@
 (defn execute-action!
   "Execute the given action with the given parameters of shape `{<parameter-id> <value>}."
   [action request-parameters]
-  (case (:type action)
-    :implicit
-    (execute-implicit-action action request-parameters)
-    (:query :http)
-    (execute-custom-action action request-parameters)
-    (throw (ex-info (tru "Unknown action type {0}." (name (:type action))) action))))
+  (let [stringified-params (update-keys request-parameters name)]
```
Comment (calherries) [inline src/metabase/actions/execution.clj:165]: It's strange to see a correction of the deserialization in the execution namespace. I would have expected it in the `defendpoint` definitions, even if it's duplicated among multiple endpoints.

Comment (tsmacdonald) [reply src/metabase/actions/execution.clj:165]: My thought was that functions should be as permissive as reasonably possible in their input

Comment (dpsutton) [reply src/metabase/actions/execution.clj:165]: This is complicated i think. The select on actions runs a normalization over parameters which keywordizes the param keys. I think ever keywordizing what are really strings is the ultimate culprit of this behavior. The ids are strings, they come specified as strings from the FE, etc. And 

```clojure
(defn normalize-parameters-list
  "Normalize `parameters` or `parameter-mappings` when coming out of the application database or in via an API request."
  [parameters]
  (or (mbql.normalize/normalize-fragment [:parameters] parameters)
      []))
``` 
ultimately that just keywordizes all keys in a map. Which i think is clunky.

Comment (calherries) [reply src/metabase/actions/execution.clj:165]: Yes, it's the keywordization of the keys in the request middleware that requires us to reverse it. But that's a quirk of request body deserialization that the execution namespace shouldn't need to worry about.

**Review by metamben — COMMENTED** [review]
I also think that reversing the keywordization should stay in the endpoint.
Shouldn't there be tests related to data permissions? Are they out of scope?

**Comment by tsmacdonald** [comment]
All right, I think I'm too much of a duck typer to fully agree with it, but I'm happy to be wrong on this and moved the stringification back to the API layer.

**Review by ranquild — APPROVED** [review]

**Pull request merged**